src/server/scripts/EasternKingdoms/ScarletMonastery/instance_scarlet_monastery.cpp

```diff
@@ -339,162 +487,202 @@ class npc_mograine : public CreatureScript
             }
         }
 
+        void ScheduleGround()
+        {
+            _scheduler.Schedule(1s, 5s, [this](TaskContext context)
+                {
+                    DoCastVictim(SPELL_CRUSADER_STRIKE);
+                    context.Repeat(10s);
+                })
+                .Schedule(6s, 11s, [this](TaskContext context)
+                    {
+                        DoCastVictim(SPELL_HAMMER_OF_JUSTICE);
+                        context.Repeat(60s);
+                    });
+        }
+
         void Reset() override
         {
-            //Incase wipe during phase that mograine fake death
-            me->RemoveUnitFlag(UNIT_FLAG_NON_ATTACKABLE);
+            me->SetReactState(REACT_AGGRESSIVE);
+            me->SetStandState(UNIT_STAND_STATE_STAND);
+            me->RemoveUnitFlag(UNIT_FLAG_IMMUNE_TO_NPC);
             me->RemoveUnitFlag(UNIT_FLAG_NOT_SELECTABLE);
-            me->RemoveAurasDueToSpell(SPELL_PERMANENT_FEIGN_DEATH);
-            SayAshbringer = false;
-            timer = 0;
-            step = 1;
-            hasDied = false;
-            heal = false;
-            fakeDeath = false;
             events.Reset();
+            _fakeDeath = false;
+            _scheduler.CancelAll();
+            if (Creature* summonedMograine = me->FindNearestCreature(NPC_HIGHLORD_MOGRAINE, 100.0f))
+            {
+                summonedMograine->DespawnOrUnsummon();
+            }
+            SayAshbringer = false;
         }
 
-        void MoveInLineOfSight(Unit* who) override
+        void EnterEvadeMode(EvadeReason /*why*/) override
         {
-            if (who && who->GetDistance2d(me) < 15.0f)
-                if (Player* player = who->ToPlayer())
-                    if (player->HasAura(AURA_ASHBRINGER) && !SayAshbringer)
-                    {
-                        me->SetFaction(FACTION_FRIENDLY);
-                        me->SetSheath(SHEATH_STATE_UNARMED);
-                        me->SetStandState(UNIT_STAND_STATE_KNEEL);
-                        me->SetFacingToObject(player);
-                        // me->Yell(12389, LANG_UNIVERSAL, player); // Doesn't exist
-                        SayAshbringer = true;
-                    }
+            if (instance)
+            {
+                instance->SetData(TYPE_MOGRAINE_AND_WHITE_EVENT, FAIL);
+            }
+        }
+
+        void MovementInform(uint32 type, uint32 id) override
+        {
+            if (type != POINT_MOTION_TYPE)
+                return;
 
-            ScriptedAI::MoveInLineOfSight(who);
+            if (id == 1)
```

Comment:
can the id be anything but one?

Reply:
Sorry, I don't understand what that means? Does it mean why the ID is 1? If this is the case, there is an event that moves towards Whitemane after completing the resurrection
case EVENT_MOVE:
```
    if (Unit* Whitemane = ObjectAccessor::GetUnit(*me, instance->GetGuidData(DATA_WHITEMANE)))
    {

        float fx, fy, fz;
        me->RemoveUnitFlag(UNIT_FLAG_NOT_SELECTABLE);
        Whitemane->GetContactPoint(me, fx, fy, fz, 0.0f);
        me->GetMotionMaster()->MovePoint(1, fx, fy, fz);
    }
    break;
```